pkg/operator/etcdcertcleaner/etcd_cert_cleaner_controller_test.go

```diff
@@ -0,0 +1,159 @@
+package etcdcertcleaner
+
+import (
+	"context"
+	"github.com/openshift/cluster-etcd-operator/pkg/operator/operatorclient"
+	u "github.com/openshift/cluster-etcd-operator/pkg/testutils"
+	"github.com/openshift/cluster-etcd-operator/pkg/tlshelpers"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/fake"
+	k8stesting "k8s.io/client-go/testing"
+	"testing"
+	"time"
+)
+
+func TestPreDeleteLabelHappyPath(t *testing.T) {
+	cleaner := newMockCleaner(t, []runtime.Object{
+		u.FakeNode("master-0", u.WithMasterLabel()),
+		u.FakeSecret(
+			operatorclient.TargetNamespace,
+			tlshelpers.GetPeerClientSecretNameForNode("master-1"),
+			map[string][]byte{}),
+	})
+	err := cleaner.sync(context.Background(), nil)
+	require.NoError(t, err)
+
+	// secret should NOT be deleted
+	updated := false
+	deleted := false
+	fx := cleaner.kubeClient.(*fake.Clientset)
+	for _, action := range fx.Actions() {
+		if action.Matches("delete", "secrets") {
+			require.Equal(t, "etcd-peer-master-1", action.(k8stesting.DeleteAction).GetName())
+			deleted = true
+		} else if action.Matches("update", "secrets") {
+			// secret should be relabeled with the grace annotation though
+			obj := action.(k8stesting.UpdateAction).GetObject().(*corev1.Secret)
+			require.Equal(t, "etcd-peer-master-1", obj.Name)
+			require.Contains(t, obj.Annotations, DeletionGracePeriodAnnotation)
+			updated = true
+		}
+	}
+
+	require.Truef(t, updated, "expected secret to be deleted, but only found actions: %v", fx.Actions())
```

Comment:
```suggestion
	require.Truef(t, updated, "expected secret to be updated, but only found actions: %v", fx.Actions())
```

Reply:
applied, thanks for double-checking. I should stop copy and pasting :rofl: